core/src/main/resources/org/apache/spark/ui/static/historypage.js

```diff
@@ -38,8 +38,17 @@ function makeIdNumeric(id) {
 }
 
 function formatDate(date) {
-  if (date <= 0) return "-";
-  else return date.split(".")[0].replace("T", " ");
+  if (date <= 0) {
+    return "-";
+  } else {
+    var dt = new Date(date);
+    return dt.getFullYear() + "-" +
```

Comment:
not familiar with js, does js have something like java `SimpleDateFormat`?

Reply:
Native Javascript does not support.
https://www.w3schools.com/js/js_date_formats.asp